benchmarking-cli/src/utils_input.rs

```diff
@@ -0,0 +1,264 @@
+use miden_vm::{
+    crypto::{MerkleStore, MerkleTree, SimpleSmt},
+    math::{Felt, FieldElement},
+    utils::collections::BTreeMap,
+    AdviceInputs, MemAdviceProvider, StackInputs, StackOutputs, Word,
+};
+
+/// The Outputs struct is used to serialize the output of the program.
+/// Via Rust WASM we cannot return arbitrary structs, so we need to serialize it to JSON.
+/// Here we need the Outputs because they can be inputs for the verifier.
+#[derive(serde::Deserialize, serde::Serialize)]
+pub struct Outputs {
+    pub stack_output: Vec<u64>,
+    pub trace_len: Option<usize>,
+    pub overflow_addrs: Option<Vec<u64>>,
+    pub proof: Option<Vec<u8>>,
+}
+
+// MERKLE DATA
+// ================================================================================================
+
+/// Struct used to deserialize merkle data from input file. Merkle data can be represented as a
+/// merkle tree or a sparse merkle tree.
+#[derive(serde::Deserialize)]
+pub enum MerkleData {
+    /// String representation of a merkle tree.  The merkle tree is represented as a vector of
+    /// 32 byte hex strings where each string represents a leaf in the tree.
+    #[serde(rename = "merkle_tree")]
+    MerkleTree(Vec<String>),
+    /// String representation of a sparse merkle tree. The sparse merkle tree is represented as a
+    /// vector of tuples where each tuple consists of a u64 node index and a 32 byte hex string
+    /// representing the value of the node.
+    #[serde(rename = "sparse_merkle_tree")]
+    SparseMerkleTree(Vec<(u64, String)>),
+}
+
+// INPUT FILE
+// ================================================================================================
+
+// TODO consider using final types instead of string representations.
+/// Input file struct that is used to deserialize input data from file. It consists of four
+/// components:
+/// - operand_stack
+/// - advice_stack
+/// - advice_map
+/// - merkle_store
+#[derive(serde::Deserialize)]
+pub struct InputFile {
+    /// String representation of the initial operand stack, composed of chained field elements.
+    pub operand_stack: Option<Vec<String>>,
+    /// Optional string representation of the initial advice stack, composed of chained field
+    /// elements.
+    pub advice_stack: Option<Vec<String>>,
+    /// Optional map of 32 byte hex strings to vectors of u64s representing the initial advice map.
+    pub advice_map: Option<BTreeMap<String, Vec<u64>>>,
+    /// Optional vector of merkle data which will be loaded into the initial merkle store. Merkle
+    /// data is represented as 32 byte hex strings and node indexes are represented as u64s.
+    pub merkle_store: Option<Vec<MerkleData>>,
+}
+
+/// Helper methods to interact with the input file
+impl InputFile {
```

Comment:
Question: We do the same on the `miden` binary, no? Can't we use that code instead of re-implementing it?

Reply:
I think this is probably copied directly from there. I agree that would be good to use a common component - but this is probably for a different PR. Let's create an issue for this.